test/marketplace.js

```diff
@@ -14,11 +14,18 @@ async function waitUntilCancelled(request) {
   await advanceTimeToForNextBlock((await currentTime()) + request.expiry + 1)
 }
 
-async function waitUntilStarted(contract, request, proof, token) {
-  await token.approve(contract.address, price(request) * request.ask.slots)
+async function waitUntilStarted(
+  contract,
+  request,
+  proof,
+  token,
+  ignoreSlotIndexes = []
+) {
+  await token.approve(contract.address, maxPrice(request) * request.ask.slots)
 
   for (let i = 0; i < request.ask.slots; i++) {
-    await contract.fillSlot(requestId(request), i, proof)
+    if (!ignoreSlotIndexes.includes(i))
```

Comment:
How can the contract be started if not all slots are filled? Maybe this needs its own function?

Reply:
Right, but the contract won't be started if the optional parameter is used to exclude a slot from being filled...

Reply:
Well yes, but then you should know what you are doing if you are specifying that optional parameter...

Reply:
🟠 I find this confusing as well. Why not add a `waitUntilSlotsFilled(indices)`? You can even use it to implement `waitUntilStarted()`